frontend/components/AddHostsModal/DownloadInstallers/DownloadInstallers.tsx

```diff
@@ -0,0 +1,184 @@
+  const [includeDesktop, setIncludeDesktop] = useState(true); // TODO: confirm default false for includeDesktop
+  const [isDownloadError, setIsDownloadError] = useState(false);
+  const [isDownloading, setIsDownloading] = useState(false);
+  const [isDownloadSuccess, setIsDownloadSuccess] = useState(false);
+  const [selectedInstaller, setSelectedInstaller] = useState<
+    IInstallerType | undefined
+  >();
+
+  const downloadInstaller = async (installerType?: IInstallerType) => {
+    if (!installerType) {
+      // do nothing
+      return;
+    }
+    setIsDownloading(true);
+    try {
+      const blob: BlobPart = await installerAPI.downloadInstaller({
+        enrollSecret,
+        installerType,
+        includeDesktop,
+      });
+      const filename = `fleet-osquery.${installerType}`;
+      const file = new global.window.File([blob], filename, {
+        type: "application/octet-stream",
+      });
+      FileSaver.saveAs(file);
+      setIsDownloadSuccess(true);
+    } catch (err) {
+      console.log(err);
+      setIsDownloadError(true);
+    } finally {
+      setIsDownloading(false);
+    }
+  };
+
+  const onClickSelector = (type: IInstallerType) => {
+    if (isDownloading) {
+      // do nothing
+    } else if (type === selectedInstaller) {
+      setSelectedInstaller(undefined);
+    } else {
+      setSelectedInstaller(type);
+    }
+  };
+
+  if (isDownloadError) {
+    return (
+      <div className={`${baseClass}__error`}>
+        <DataError />
+      </div>
+    );
+  }
+
+  if (isDownloadSuccess) {
+    const installerPlatform =
+      (selectedInstaller &&
+        `${INSTALLER_PLATFORM_BY_TYPE[selectedInstaller]} `) ||
+      "";
+    return (
+      <div className={`${baseClass}__success`}>
+        <img src={SuccessIcon} alt="download successful" />
+        <h2>You&rsquo;re almost there</h2>
+        <p>{`Run the installer on a ${installerPlatform}laptop, workstation, or sever to add it to Fleet.`}</p>
+        <Button onClick={onCancel}>Got it</Button>
+      </div>
+    );
+  }
+
+  return (
+    <div className={`${baseClass}`}>
+      <p>Which platform is your host running?</p>
+      <div className={`${baseClass}__select-installer`}>
+        {displayOrder.map((platform) => {
+          const installerType = INSTALLER_TYPE_BY_PLATFORM[platform];
+          const isSelected = selectedInstaller === installerType;
+          return (
+            <div
+              key={installerType}
+              className={`${baseClass}__selector ${
+                isSelected ? `${baseClass}__selector--selected` : ""
+              }`}
+              onClick={() => onClickSelector(installerType)}
+            >
+              <span>
+                {displayIcon(platform, isSelected)}
+                {platform}
+              </span>
+            </div>
+          );
+        })}
+      </div>
+      <Checkbox
+        name="include-fleet-desktop"
+        onChange={() => setIncludeDesktop(!includeDesktop)}
```

Comment:
@lukeheath I believe that `onChange` requires a function to call rather than an evaluated function

Reply:
Actually the `onChange` function signature is `any` so that's not super helpful. But it does expect a function to be called with a value provided by the caller. I followed the pattern that was used in the other version of the modal, which looks like it circumvents the control of the value. It appears that the net effect is the same but it looks like the intended usage of the `Checkbox` component is `onChange={(value) => setIncludeDesktop(value)}` since the `Checkbox` component will pass the inverted value into the `onChange` function.